File: OOAD/shopping-cart-app/src/shopping/components/Order.java
package shopping.components;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;


public class Order {
	private static int totalNumberOfid;
	private int id;
	private List<LineItem> items=new ArrayList<LineItem>();
	Date date=new Date();
	private double totalPrice=0;
	LineItem lineItem;
	Product product;
	
	static{
		totalNumberOfid=0;
	}
	
	public Order(Date date){
		this.date=date;
		id=generateId();
	}
	
	public void addLineItem(LineItem item){
		for(LineItem lineItem:items){
			if(lineItem.getProduct().getProductId()==item.getProduct().getProductId()){
				lineItem.addQuantity(item.getLineItemQuantity());
			}
		}
		items.add(item);
	}
	
	public ArrayList<LineItem> getLineItemsList(){
		return (ArrayList<LineItem>) items;
	}
	
	public double checkOutPrice(){
		//LineItem lineItem=new LineItem();
		for(LineItem item:items){
			totalPrice+= item.getLineItemCost();
		}
		return totalPrice;
	}
	
	public int generateId(){
		return (++totalNumberOfid);
	}
	
	public int getOrderId(){
		return id;
	}
	
	public Date getDate(){
		return date;
	}
	
}

File: C#/OOP/ShopingCart-App/ShopingCart-App/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ShopingCart_App
{
    class Order
    {
        private static int totalNumberOfid;
        private int id;
        private List<LineItem> items = new List<LineItem>();
        DateTime date = new DateTime();
        private double totalPrice = 0;
        LineItem lineItem;
        Product product;

        public int OrderId
        {
            get
            {
                return id;
            }

            set
            {
                id = value;
            }
        }

        public DateTime OrderDate
        {
            get
            {
                return date;
            }

            set
            {
                date = value;
            }
        }

        static Order(){
		totalNumberOfid=0;
	}

    public Order(DateTime date)
    {
        this.OrderDate = date;
        OrderId = GenerateId();
    }

    public void AddLineItem(LineItem item)
    {
        foreach (LineItem lineItem in items)
        {
            if (lineItem.getProduct().ProductId == item.getProduct().ProductId)
            {
                lineItem.addQuantity(item.Quantity);
            }
        }
        items.Add(item);
    }

    public List<LineItem> GetLineItemsList()
    {
        return (List<LineItem>)items;
    }

    public double CheckOutPrice()
    {
        //LineItem lineItem=new LineItem();
        foreach (LineItem item in items)
        {
            totalPrice += item.getLineItemCost();
        }
        return totalPrice;
    }

    public int GenerateId()
    {
        return (++totalNumberOfid);
    }
}
}
